### PR TITLE
Added restart server command

### PR DESCRIPTION
**Pull request by TheUbMunster**
"restartserver" restarts the server in-place (doesn't close & reopen console window, but *does* logically restart the program.)

### CODE REVIEW

**Comment by TheUbMunster** [comment]
pending testing on non-windows

File: Server/Program.cs [hunk 1]
```diff
@@ -542,6 +543,21 @@ async Task LoadShines()
...
+    if (args.Length != 0)
+    {
+        return "Usage: restartserver (no arguments)";
+    }
+    else
+    {
+        consoleLogger.Info("Received restartserver command");
+        restartRequested = true;
+        cts.Cancel();
+        return "restarting...";
```
Comment (piplup55) [inline Server/Program.cs]: i would recommand changing "restarting" to "Restarting"

File: Server/Program.cs [hunk 2]
```diff
@@ -565,4 +581,16 @@ async Task LoadShines()
...
\ No newline at end of file
+await listenTask;
+if (restartRequested) //need to do this here because this needs to happen after the listener closes, and there isn't an
+                      //easy way to sync in the restartserver command without it exiting Main()
+{
+    string? path = System.Reflection.Assembly.GetEntryAssembly()?.GetName().Name;
+    const string unableToStartMsg = "Unable to ascertain the executable location, you'll need to re-run the server manually.";
+    if (path != null) //path is probably just "Server", but in the context of the assembly, that's all you need to restart it.
+    {
+        Console.WriteLine($"Running (pid): {System.Diagnostics.Process.Start(path)?.Id.ToString() ?? unableToStartMsg}");
```
Comment (piplup55) [inline Server/Program.cs]: i would change this from "Running (pid)" to "Server Running on (pid)"

**Review by piplup55 — APPROVED** [review]

**Review by Sanae6 — APPROVED** [review]
While I do approve of this solution, and I see the reason for this implementation, I also don't particularly like the idea of restarting the server internally. This feels like something which should kill the server and depend on an external process to restart it, whether that be pm2, docker, or our own wrapper process.

**Comment by TheUbMunster** [comment]
> While I do approve of this solution, and I see the reason for this implementation, I also don't particularly like the idea of restarting the server internally. This feels like something which should kill the server and depend on an external process to restart it, whether that be pm2, docker, or our own wrapper process.

Probably best to do as you say, however, I found it useful (and it was quite simple to write this way) as the purpose of this was to be able to restart the server remotely whenever buggy stuff happens. e.g., couple of days ago testing on build that piplup was hosting, but he was away for awhile so if something went wrong I could restart it myself with a simple command. If you'd rather, I can make the command start up a second process, kill the server, then have the second process restart the server, then kill itself.

**Comment by Sanae6** [comment]
I think it's still fine that it's in the server, others who don't use the solutions I mentioned can simply use this.

**Pull request merged**